pubsub/google/cloud/pubsub_v1/publisher/batch/thread.py

```diff
@@ -234,9 +236,10 @@ def monitor(self):
 
         # Sleep for however long we should be waiting.
         time.sleep(self._settings.max_latency)
-
         _LOGGER.debug('Monitor is waking up')
-        return self._commit()
+
+        if self._size:
+            return self._commit()
```

Comment:
This is potentially racy (e.g., if another thread calls `_commit()`.  Could we just make `_commit` bail early in this case?  Maybe drop the warning message, too.

Reply:
you are right.  It actually does bail early already.